src/Rule/Nested.php

```diff
@@ -186,20 +189,39 @@ private static function ensureArrayHasRules(iterable &$rules)
 
     private function normalizeRules(): void
     {
-        /** @var iterable $rules */
+        if ($this->rules === null) {
+            return;
+        }
+
+        /** @var iterable<RuleInterface> $rules */
         $rules = $this->rules;
+        if ($rules instanceof Traversable) {
+            $rules = iterator_to_array($rules);
+        }
 
         while (true) {
             $breakWhile = true;
             $rulesMap = [];
 
+            /** @var mixed $valuePath */
```

Comment:
Technically no, at this point it can contain other types, such as float, if user defined them. The inner check in the loop makes sure it's only int or string. 

Need to add a relevant test though. But I was planning to improve code coverage in a separate PR.